### PR TITLE
Mavlink

### PR DESCRIPTION
**Pull request by MarwanMashaly1**
Implementation of Mavlink in proxy to read from serial.

This is based from #5 so probably should merge it first

### CODE REVIEW

File: hydra_provider/Cargo.toml [hunk 1]
```diff
@@ -6,14 +7,19 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-messages = { git = "https://github.com/uorocketry/hydra", branch = "master" }
+messages = { git = "https://github.com/uorocketry/hydra", branch = "messages-improvements", features = ["ts"] }
+mavlink = { git = "https://github.com/uorocketry/rust-mavlink", branch = "hydra_dialect", features = ["default"] }
```
Comment (NoahSprenger) [inline hydra_provider/Cargo.toml:11]: Should we merge hydra_dialect to master of the rust-mavlink fork?

File: hydra_provider/Cargo.toml [hunk 2]
```diff
@@ -6,14 +7,19 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-messages = { git = "https://github.com/uorocketry/hydra", branch = "master" }
+messages = { git = "https://github.com/uorocketry/hydra", branch = "messages-improvements", features = ["ts"] }
```
Comment (NoahSprenger) [inline hydra_provider/Cargo.toml:10]: This should be changed to master as messages-improvements was merged.

File: hydra_provider/tests/radio_message.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+fd 58 00 00 2a 01 01 02 00 00 07 fe f4 63 01 01 09 19 72 d4 be bc 08 dd 56 3d 20 89 1c c1 02 77 21 c0 8f 46 8d 3f f6 4c bf 42 28 34 c4
```
Comment (NoahSprenger) [inline hydra_provider/tests/radio_message.txt:1]: This should be removed.

File: hydra_provider/Cargo.toml (hunk 1 above)
Comment (MarwanMashaly1) [reply hydra_provider/Cargo.toml:11]: not yet, once we are sure of our dialect

File: hydra_provider/src/input/serial_input.rs [hunk 4]
```diff
@@ -26,29 +32,83 @@ impl SerialInput {
...
+    }
+}
 
-        let port = serialport::new(&port, baud_rate)
-            .timeout(Duration::new(30, 0))
-            .open()
-            .with_context(|| format!("Failed to open serial connection '{port}'"))?;
+impl SerialInput {
+    pub fn read_write_loop(&self, send: Sender<InputData>) -> ! {
+        thread::scope(|s| {
```
Comment (jonapap) [inline hydra_provider/src/input/serial_input.rs:44]: If anyone is confused on how I'm able to send and receive from two separate threads without angering the borrow checker, let me know, and I can explain a bit more what is happening here.

**Pull request merged**